### ISSUE TITLE
sub class method can not complete use parent class phpdoc

### ISSUE THREAD

**Issue by manaphp** [klesun/deep-assoc-completion#125]
<?php

class Controller
{
    /**
     * @return array
     */
    private function helpGetAcl()
    {
        return get_class_methods(static::class);
    }

    /**
     * @return array =static::helpGetAcl()
     */
    abstract public function getAcl();
}

class TestController extends Controller
{

    public function getAcl()
    {
        return ['CARET'] // not complete
    }

    public function indexAction()
    {

    }

    public function deleteAction()
    {

    }
}

**Comment by manaphp** [comment]
<?php

class Controller
{
    /**
     * @return array
     */
    private function helpGetAcl()
    {
        return get_class_methods(static::class);
    }

    /**
     * @return array =static::helpGetAcl()
     */
    abstract public function getAcl();
}

class TestController extends Controller
{

    public function getAcl()
    {
        return ['CARET'] // not complete
    }

    public function indexAction()
    {

    }

    public function deleteAction()
    {

    }
}

**Comment by klesun** [comment]
Oh, it actually a step towards the https://github.com/klesun/deep-assoc-completion/issues/127

So the task is to add `get_class_methods()`, sounds reasonable, sorry for ignoring this for two year...

**Comment by manaphp** [comment]
> Oh, it actually a step towards the #127
> 
> So the task is to add `get_class_methods()` support similarly to `get_class_vars()`, sounds reasonable, sorry for ignoring this for two years...

add php builtin function supported needs plugin do special work?

**Comment by manaphp** [comment]
> > Oh, it actually a step towards the #127
> > So the task is to add `get_class_methods()` support similarly to `get_class_vars()`, sounds reasonable, sorry for ignoring this for two years...
> 
> add php builtin function supported needs plugin do special work?

can let plugin-user to define these functions?

**Comment by klesun** [comment]
Yeah, it has to be coded in the plugin itself since functions like that do the meta stuff that you can't achieve with normal php code. I actually had [a campaign](https://github.com/klesun/deep-assoc-completion/issues/70) to cover all such built-in functions, but apparently just plainly missed the `get_class_methods()`.

**Comment by klesun** [comment]
Btw, related issue: https://github.com/klesun/deep-assoc-completion/issues/97
